x-pack/packages/kbn-slo-schema/src/rest_specs/slo.ts

```diff
@@ -182,23 +190,21 @@ const fetchHistoricalSummaryResponseSchema = t.array(
   })
 );
 
-/**
- * The query params schema for /internal/observability/slo/_definitions
- *
- * @private
- */
-const findSloDefinitionsParamsSchema = t.type({
-  query: t.type({
+const findSloDefinitionsParamsSchema = t.partial({
+  query: t.partial({
     search: t.string,
+    includeOutdatedOnly: toBooleanRt,
+    page: t.string,
+    perPage: t.string,
   }),
 });
 
-/**
- * The response schema for /internal/observability/slo/_definitions
- *
- * @private
- */
-const findSloDefinitionsResponseSchema = t.array(sloResponseSchema);
+const findSloDefinitionsResponseSchema = t.type({
```

Comment:
More generally, I want to go through all these files and refactor them. I think the project grew, and we lost control on the schema/rest_specs/models. I think it would be good to spend some time thinking how to structure all these types.
Maybe having smaller building blocks, that can be reused to build the rest_specs and the models.
I think it's worth opening an issue for this

Reply:
https://github.com/elastic/kibana/issues/172617